cpg-core/src/main/java/de/fraunhofer/aisec/cpg/passes/scopes/LoopScope.kt

```diff
@@ -38,7 +38,26 @@ class LoopScope(loopStatement: Statement) :
     var starts: List<Node> = ArrayList()
 
     /** Statements that constitute the start of the Loop condition evaluation, mostly of size 1 */
-    var conditions: List<Node> = ArrayList()
+    val conditions: List<Node>
+        get() =
+            when (val node = astNode) {
+                is WhileStatement ->
+                    mutableListOf(node.condition, node.conditionDeclaration).filterNotNull()
+                is ForStatement -> mutableListOf(node.condition).filterNotNull()
+                is ForEachStatement -> mutableListOf(node.variable).filterNotNull()
```

Comment:
This should be the iterator, however, the start of the loop is defined by the EOGPass and retrieved from the EOG path edges.